src/html/document.zig

```diff
@@ -81,14 +84,18 @@ pub const HTMLDocument = struct {
         }
     }
 
-    // TODO: not implemented by libdom
-    pub fn get_cookie(_: *parser.DocumentHTML) ![]const u8 {
-        return error.NotImplemented;
+    pub fn get_cookie(_: *parser.DocumentHTML, allocator: std.mem.Allocator, userctx: UserContext) ![]const u8 {
+        var buf: std.ArrayListUnmanaged(u8) = .{};
+        defer buf.deinit(allocator);
+        try userctx.cookie_jar.forRequest(&userctx.url.uri, buf.writer(allocator), .{ .navigation = true });
+        return buf.toOwnedSlice(allocator);
```

Comment:
i know this is consistent with a lot of the existing APIs, but if you _know_ that allocator is an arena, that this an unnecessary/extra allocation + copy. Plus, we never free the dupe, so it's a bit like the worst of both: don't assume it's an arena, but also don't free the allocation.

There are comments like https://github.com/lightpanda-io/browser/blob/main/src/url/url.zig#L75  which imply this should be fixed. But I think relying on the page_arena is fine (though, we should rename the allocator parameter `arena`, IMO)

Reply:
I agree, I changed.